packages/gatsby-source-filesystem/src/create-file-node.js

```diff
@@ -20,8 +21,24 @@ exports.createFileNode = async (pathToFile, pluginOptions = {}) => {
     absolutePath: slashed,
   }
   // console.log('createFileNode', slashedFile.absolutePath)
-  const contentDigest = await md5File(slashedFile.absolutePath)
   const stats = await fs.stat(slashedFile.absolutePath)
+  const isDirectory = stats.isDirectory()
+  var internal;
+  if(isDirectory) {
+    const contentDigest = await md5Dir(slashedFile.absolutePath)
```

Comment:
This isn't necessary. It's not interesting to know for the directory node if a file within the directory changed. Instead you should make a digest of the directory stats object and anything else we add to the Directory node.